bangolufsen/media_player.py

```diff
@@ -96,24 +134,29 @@ def state(self):
         else:
             return STATE_OFF
 
+# Turn on uses volume up which for most devices, turns it on without changing source, but it does nothing on the BeoSound system.
     def turn_on(self):
         self._pwon = True
-        self.select_source(DEFAULT_SOURCE)
```

Comment:
If you don't have a device like the BeoSound that can maintain the source state when the speakers are turned off, they will default to Radio when turned on if using Volume Up. I don't have a BeoSound and therefore need to select the DVD source for my audio to go through. Any suggestions on how to deal with this?

Reply:
the best solution is probably then to give a configuration option with the default source (DVD or A.MEM) and the user can set it. Also the problem of turning on with volume up is that the plugin then doesn't know the state of the actual source (because mlgw does not send status messages for all sources) so you end up inconsistent between plugin and actual state of the b&o audio system. So all in all probably better option is to turn on by selecting a default source.

Reply:
Ok, so I fixed this now. The default source (e.g. DVD) and the list of available sources (e.g., CD, DVD, RADIO) is now configurable. Also the Turn on command always uses select source. I think it's ready to go!

Reply:
Looks great!